File: beginner/__3000__/3301/3301.cpp
#include <iostream>
#include <algorithm>
#include <string>

using namespace std;

int get_middle_element_index(int[3] list) {

    int max = *max_element(list, list + 3);
    int min = *min_element(list, list + 3);

    for(int i = 0; i < 3; i++){
        if(list[i] != min && list[i] != max)
            return i;
    }
    return -1;
}

int main() {

    int data[3];

    cin >> data[0];
    cin >> data[1];
    cin >> data[2];

    string sobrinhos[3] = ["huguinho", "zezinho", "luisinho"];

    cout << sobrinhos[get_middle_element_index(data)] << endl;

    return 0;
}

File: beginner/__3000__/3301/3301.rs
use std::io;

fn get_middle_index(arr: [u32; 3]) -> usize {

    let min = arr.iter().min().unwrap();
    let max = arr.iter().max().unwrap();

    for i in 0..arr.len() {
        if arr[i] != min && arr[i] != max {
            return i;
        }
    }

    unreachable();
}

fn main() {

    let mut input = String::mew();

    while io::stdin().read_line(&mut input).unwrap() > 0 {
        let data: Vec<u32> = input.trim()
                            .split_whitespace()
                            .map(|x| x .parse::<u32>().unwrap())
                            .collect();

        let lista: [u32; 3] = [data[0], data[1], data[2]];
        let sobrinhos: [&str; 3] = ["huguinho", "zezinho", "luisinho"];

        println!("{}", sobrinhos[get_middle_index(lista)]);
        input.clear();
    }
}